modules/local-caching-classloader/src/test/java/org/apache/accumulo/classloader/lcc/MiniAccumuloClusterClassLoaderFactoryTest.java

```diff
@@ -0,0 +1,284 @@
+
+    Path baseDirPath = tempDir.resolve("base");
+    Path jsonDirPath = baseDirPath.resolve("contextFiles");
+    Files.createDirectory(jsonDirPath, PERMISSIONS);
+
+    // Create a context definition that only references jar A
+    final ContextDefinition testContextDef =
+        ContextDefinition.create("test", MONITOR_INTERVAL_SECS, jarAOrigLocation);
+    final String testContextDefJson = testContextDef.toJson();
+    final File testContextDefFile = jsonDirPath.resolve("testContextDefinition.json").toFile();
+    Files.writeString(testContextDefFile.toPath(), testContextDefJson, StandardOpenOption.CREATE);
+    assertTrue(Files.exists(testContextDefFile.toPath()));
+
+    final String[] names = this.getUniqueNames(1);
+    try (AccumuloClient client =
+        Accumulo.newClient().from(getCluster().getClientProperties()).build()) {
+
+      List<String> tservers = client.instanceOperations().getTabletServers();
+      Collections.sort(tservers);
+      assertEquals(3, tservers.size());
+
+      final String tableName = names[0];
+
+      final IngestParams params = new IngestParams(client.properties(), tableName, 100);
+      params.cols = 10;
+      params.dataSize = 10;
+      params.startRow = 0;
+      params.columnFamily = "test";
+      params.createTable = true;
+      params.numsplits = 3;
+      params.flushAfterRows = 0;
+
+      TestIngest.createTable(client, params);
+
+      // Confirm 4 tablets, spread across 3 tablet servers
+      client.instanceOperations().waitForBalance();
+
+      final List<TabletMetadata> tm = getLocations(((ClientContext) client).getAmple(),
+          client.tableOperations().tableIdMap().get(tableName));
+      assertEquals(4, tm.size()); // 3 tablets
+
+      final Set<String> tabletLocations = new TreeSet<>();
+      tm.forEach(t -> tabletLocations.add(t.getLocation().getHostPort()));
+      assertEquals(3, tabletLocations.size()); // 3 locations
+
+      // both collections are sorted
+      assertIterableEquals(tservers, tabletLocations);
+
+      TestIngest.ingest(client, params);
+
+      final VerifyParams vp = new VerifyParams(client.properties(), tableName, params.rows);
+      vp.cols = params.cols;
+      vp.rows = params.rows;
+      vp.dataSize = params.dataSize;
+      vp.startRow = params.startRow;
+      vp.columnFamily = params.columnFamily;
+      vp.cols = params.cols;
+      VerifyIngest.verifyIngest(client, vp);
+
+      // Set the table classloader context. Context name is the URL to the context
+      // definition file
+      final String contextURL = testContextDefFile.toURI().toURL().toString();
+      client.tableOperations().setProperty(tableName, Property.TABLE_CLASSLOADER_CONTEXT.getKey(),
+          contextURL);
+
+      // Attach a scan iterator to the table
+      IteratorSetting is = new IteratorSetting(101, "example", ITER_CLASS_NAME);
+      client.tableOperations().attachIterator(tableName, is, EnumSet.of(IteratorScope.scan));
```

Comment:
Setting this is eventually consistent, so its possible the scan may not see the setting.  Could cause intermittent failures.  Could retry the scan until the count is 1000.